server/handlers/default.go

```diff
@@ -20,6 +25,69 @@ import (
 	"github.com/docker/notary/tuf/validation"
 )
 
+// NewCacheControlConfig creates a new configuration for Cache-Control headers,
+// which by default, sets cache max-age values for consistent (by checksum)
+// downloads 30 days and non-consistent (current) downloads to 5 minutes
+func NewCacheControlConfig() *CacheControlConfig {
+	return &CacheControlConfig{
+		headerVals: map[string]int{
+			"consistent": 30 * 24 * 60 * 60, // 30 days
+			"current":    5 * 60,            // 5 minutes
+		},
+	}
+}
+
+// CacheControlConfig is the configuration for the max cache age for
+// cache control headers.
+type CacheControlConfig struct {
+	headerVals map[string]int
+}
+
+// SetConsistentCacheMaxAge sets the Cache-Control header value for consistent
+// downloads
+func (c *CacheControlConfig) SetConsistentCacheMaxAge(value int) {
+	c.headerVals["consistent"] = value
+}
+
+// SetCurrentCacheMaxAge sets the Cache-Control header value for current
+// (non-consistent) downloads
+func (c *CacheControlConfig) SetCurrentCacheMaxAge(value int) {
+	c.headerVals["current"] = value
+}
+
+// UpdateConsistentHeaders updates the given Headers object with the Cache-Control
+// headers for consistent downloads
+func (c *CacheControlConfig) UpdateConsistentHeaders(headers http.Header, lastModified time.Time) {
+	c.updateHeaders(headers, lastModified, true)
+}
+
+// UpdateCurrentHeaders updates the given Headers object with th eCache-Control
+// headers for current (non-consistent) downloads
+func (c *CacheControlConfig) UpdateCurrentHeaders(headers http.Header, lastModified time.Time) {
+	c.updateHeaders(headers, lastModified, false)
+}
+
+func (c *CacheControlConfig) updateHeaders(headers http.Header, lastModified time.Time, consistent bool) {
+	var seconds int
+	var cacheHeader string
+
+	if consistent {
+		seconds = c.headerVals["consistent"]
+		cacheHeader = fmt.Sprintf("public, max-age=%v, s-maxage=%v, must-revalidate", seconds, seconds)
+	} else {
+		seconds = c.headerVals["current"]
+		cacheHeader = fmt.Sprintf("public, max-age=%v, s-maxage=%v", seconds, seconds)
+	}
+
+	if seconds > 0 {
```

Comment:
It might be worth documenting somewhere that `0` time will mean that we don't cache and set the headers appropriately